launch_ros/launch_ros/utilities/evaluate_parameters.py

```diff
@@ -69,9 +69,17 @@ def check_sequence_type_is_allowed(sequence):
             if isinstance(value[0], Substitution):
                 # Value is a list of substitutions, so perform them to make a string
                 evaluated_value = perform_substitutions(context, list(value))
-                yaml_evaluated_value = yaml.safe_load(evaluated_value)
-                if yaml_evaluated_value is None:
-                    yaml_evaluated_value = ''
```

Comment:
Removing this check breaks existing launch files in Foxy (and Galactic). For example, the following used to work:

```python
import launch
from launch.actions import DeclareLaunchArgument
from launch.substitutions import LaunchConfiguration
from launch_ros.actions import Node


def generate_launch_description():
    return launch.LaunchDescription([
        DeclareLaunchArgument('foo', default_value=''),
        Node(
            package='demo_nodes_py',
            executable='talker',
            name='talker_node',
            output='screen',
            parameters = [
                {'foo': LaunchConfiguration('foo')}
            ]
        ),
    ])
```

because the `LaunchConfiguration` substitution evaluated to `None` which used to translate to the empty string.

I think we can easily fix the issue in Foxy and Galactic by adding back L73-74 above, but I'm not sure what the correct behavior should be on Rolling.

@adityapande-1995 @DLu thoughts?

cc/ @Yadunund

Reply:
See https://github.com/ros2/launch_ros/pull/300 for a proposed fix.